chasm/lib/scheduler/library.go

```diff
@@ -0,0 +1,70 @@
+package scheduler
+
+import "go.temporal.io/server/chasm"
+
+type (
+	library struct {
+		chasm.UnimplementedLibrary
+
+		generatorTaskExecutor            *GeneratorTaskExecutor
+		invokerExecuteTaskExecutor       *InvokerExecuteTaskExecutor
+		invokerProcessBufferTaskExecutor *InvokerProcessBufferTaskExecutor
+		backfillerTaskExecutor           *BackfillerTaskExecutor
+	}
+)
+
+var Library = &library{}
+
+func (l *library) SetGeneratorTaskExecutor(executor *GeneratorTaskExecutor) {
```

Comment:
nit: no quite following why we need those Setter methods instead of a constructor function or use fx.In?